Makefile

```diff
@@ -65,42 +58,42 @@ gum:
 .PHONY: install-env
 install-env:
 	cp -n .env.sample .env; \
-	ln -sf ${PWD}/.env ${PWD}/packages/web/.env.development; \
-	ln -sf ${PWD}/.env ${PWD}/packages/api/.env; \
-	ln -sf ${PWD}/.env ${PWD}/packages/services/url_shortener/.env; \
-	ln -sf ${PWD}/.env ${PWD}/packages/py-api/.env; \
-	ln -sf ${PWD}/.env ${PWD}/packages/services/questionnaire;
+	ln -sf ${PWD}/.env ${PWD}/services/web/.env.development; \
+	ln -sf ${PWD}/.env ${PWD}/services/url_shortener/.env; \
+	ln -sf ${PWD}/.env ${PWD}/services/py_api/.env; \
+	ln -sf ${PWD}/.env ${PWD}/services/questionnaire;
+
 
 .PHONY: install-python
 install-python:
-	if [ $(shell uname -s) = Linux ]; \
-	then \
-		echo 'export PATH="/$(shell whoami)/.local/bin:$$PATH"' >> ~/.zshrc; \
-		curl -sSL https://install.python-poetry.org | python3 - ;\
-		cd ./packages/py-api && \
-		PATH=/$(shell whoami)/.local/bin:$$PATH poetry install; \
+	if [ $(shell command -v apt-get) ]; then \
+		sudo apt-get update; \
+		sudo apt-get install -y python3.12; \
+		sudo apt install -y pipx; \
+	elif [ $(shell command -v brew) ]; then \
+		brew install python@3.12; \
+		brew install pipx; \
 	else \
-		echo 'export PATH="/Users/$(shell whoami)/.local/bin:$$PATH"' >> ~/.zshrc; \
-		curl -sSL https://install.python-poetry.org | python3 - ;\
-		cd ./packages/py-api && \
-		PATH=/Users/$(shell whoami)/.local/bin:$$PATH poetry install; \
-	fi \
+		echo "No supported package managers found."; \
+		exit 1; \
+	fi; \
+	pipx ensurepath; \
+	pipx install poetry; \
+	poetry config virtualenvs.in-project true; \
+	cd ./services/py_api/; \
+	poetry install; \
+	poetry run pre-commit install;
```

Comment:
How does this work? @Aleks1908 set up husky which is a pre-commit hook manager.

Would this overwrite the pre-commit hook which is already in place?
I think I had figured out something as to account for both python and node environments in the project. You can get inspired by it. 

https://github.com/AUBGTheHUB/monolith/blob/05c73aaf1a48934f224858e0dcd8c503ff9c83aa/.husky/pre-commit#L5

Reply:
Yeah I see. This one was supposed to set up the pre-commit hook for python.